apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -366,16 +369,38 @@ export default class CoreLibrary {
     return speechBubbles[speechBubbles.length - 1];
   }
 
+  // When the total number of sprites reaches MAX_NUM_SPRITE_WARNING,
+  // a workspace alert warning is displayed.
+  // MAX_NUM_SPRITE_WARNING = MAX_NUM_SPRITES - 1
+  reachedSpriteLimit() {
```

Comment:
This function reads like a boolean function, but has a side effect of triggering a warning on the code editor. This can lead to confusion when maintaining the code because someone in the future might copy what looks like a simple boolean function and get unintended side effects.

This can be fixed by splitting this into two functions. Something like `reachedSpriteLimit` and `warnIfAtSpriteLimit`.

Reply:
Thanks Jessie! This is great feedback.
I updated reachedSpriteLimit so it just checks if the total number of sprites >= MAX_NUM_SPRITES.
Then added a warnIfAtSpriteLimit to dispatch a warning if the total number of sprites == SPRITE_LIMIT_WARNING. I renamed the constant to reflect its purpose more clearly. 
I think that separating these 2 functions also clarifies the purpose of the 2 constants as well to address your comment below about the use of 2 constants instead of 1. Lmk what you think of this update.

I added a couple more unit tests to check to make sure the alert is dispatched correctly - a test to check warning is dispatched only once even when the addSprites function is called additional times past the SPRITE_LIMIT_WARNING. And then a test to make sure the alert is not dispatched at the edge case.

Thanks!